src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -2573,20 +2573,61 @@ trait Types
 
     private var trivial: ThreeValue = UNKNOWN
     override def isTrivial: Boolean = {
-      if (trivial == UNKNOWN) trivial = fromBoolean(isTrivialResult && areTrivialParams(params))
+      if (trivial == UNKNOWN) trivial = fromBoolean(isTrivialResult && areTrivialParams)
       toBoolean(trivial)
     }
 
     private def isTrivialResult =
       resultType.isTrivial && (resultType eq resultType.withoutAnnotations)
 
-    private def areTrivialParams(ps: List[Symbol]): Boolean = ps match {
-      case p :: rest =>
-        p.tpe.isTrivial && !typesContain(paramTypes, p) && !(resultType contains p) &&
-        areTrivialParams(rest)
-      case _ =>
-        true
-    }
+    private def areTrivialParams: Boolean =
+      if (params.isEmpty) true else {
+        val len = params.length
+        val paramsTpes: Array[Type] = new Array[Type](len)
+
+        // returns the result of ```params.forall(_.tpe.isTrivial))```
+        // along the way, it loads each param' tpe into array
+        def forallIsTrivial: Boolean = {
+          var res = true
+          var pps = params
+          var ix = 0
+          while(res && ix < len){
+            paramsTpes(ix) = pps.head.tpe
+            res = paramsTpes(ix).isTrivial
+            pps = pps.tail
+            ix = ix + 1
```

Comment:
Let's just use `ix += 1` here and below, which is the style we typically use elsewhere.